Update 03-large-files.md (#8760)
pantheon-systems/documentation: source/content/guides/filesystem/03-large-files.md

## Highly Populated Directories

Directories that house a large number of files can cause performance issues. Notably:

- Around 50,000 files in one directory may lead to performance degradation.
- Exceeding 100,000 files in one directory can result in very poor performance and eventually filesystem corruption, potentially causing extended site downtime and file loss.

To optimize performance:

- If possible, restructure directories that contain more than 50,000 files.
- Consider offloading files to dedicated external storage solutions, such as Amazon S3 or Google Cloud Storage.

> - Use content management settings in both WordPress and Drupal to organize uploaded content into directories based on the YYYY/MM or YYYY/MM/DD format or specific user criteria. This approach is preferable over consolidating all uploads into a single directory. For sites already facing this issue, the solution typically involves reorganizing the files into multiple directories and then updating the respective file tables in Drupal or WordPress.

Reviewer: ```suggestion
- Use content management settings in both WordPress and Drupal to organize uploaded content into directories based on the YYYY/MM or YYYY/MM/DD format or specific user criteria. This approach is preferable over writing all uploads to a single directory. For sites already facing this issue, the solution typically involves reorganizing the files into multiple directories and then updating the respective file tables in Drupal or WordPress.
```

> Useful tools for offloading are detailed below under CDNs

Reviewer: ```suggestion
Useful tools for offloading are detailed below under CDNs.
```

## Large Files

> The [Pantheon Filesystem](/guides/filesystem) and file serving infrastructure are not optimized to store and deliver large files.

Reviewer: ```suggestion
The [Pantheon Filesystem](/guides/filesystem) and file serving infrastructure are not optimized to store and deliver large files.  The maximum size for any single file is 256 MiB.
```
Documenting the hard limit here may be redundant to the table below, but IMO it's better to call this out clearly (& repeatedly? :P ).

|:----------------|--------------------------------------|-------------------------------------|
| ≤ 100MiB        | <span  style="color:green">✔</span>  | Can be uploaded via any means       |
| 100MiB - 256MiB | <span  style="color:orange">✔</span> | Must be uploaded over SFTP or rsync |

> | > 256MiB        | <span  style="color:red">❌</span>    | Must be offloaded to an external filesystem|

Reviewer: ```suggestion
| > 256MiB        | <span  style="color:red">❌</span>    | Must be offloaded to an external filesystem
```
(minor) We aren't terminating other rows with `|`; recommend staying consistent.

### CDNs

> For hosting either large files or a very large number of  media files, we recommend external file systems like Amazon S3 or Google Cloud Storage. These offer a cost-effective solution and can allow direct uploads to an external service without utilizing the Pantheon Filesystem..

Reviewer: ```suggestion
For hosting either large files or a very large number of media files, we recommend external file systems like Amazon S3 or Google Cloud Storage. These offer a cost-effective solution and can allow direct uploads without utilizing the Pantheon Filesystem.
```

> - Drupal sites can use a modules such as [S3 File System](https://www.drupal.org/project/s3fs).

Reviewer: ```suggestion
- Drupal sites can use a module such as [S3 File System](https://www.drupal.org/project/s3fs).
```

### Upload Speed

If you're transferring large files over a slow internet connection, the upload might exceed our 59-second [Connection Timeout](/timeouts/#timeouts-that-are-not-configurable). Use an [upload time calculator](https://downloadtimecalculator.com/Upload-Time-Calculator.html) to ascertain if your connection speed might prevent uploads from completing in time.

### Media and Email

Refer to the [Media and Email](/guides/platform-considerations/media-email-support) section of the [Platform Considerations](/guides/platform-considerations) guide for more information on support for large media and email files.

## Large (>100GiB) File Backups

> Large backups take longer, use more resources, and have a higher likelihood of failing. Sites exceeding 200GiB or containing over two million files can't be backed up (though code and databases are backed up as normal). Even without backups, file content is reliably stored across multiple servers.

Reviewer: ```suggestion
Large backups take longer, use more resources, and have a higher likelihood of failing.
Filesystems exceeding 200 GiB or containing over two million files cannot be backed up (though code and databases are backed up as normal).
Even without filesystem backups, file content is protected by redundant storage.
```
"multiple servers" feels archaic in a cloud environment - how about using "redundant storage" instead?
